src/categorical_algebra/DataMigrations.jl

```diff
@@ -438,9 +383,24 @@ function colimit_representables(F::DeltaSchemaMigration, y)
 end
 function colimit_representables(F::ConjSchemaMigration, y)
   C = dom(F)
+  ACS = constructor(ob_map(y,first(ob_generators(dom(y)))))
   colimits = make_map(ob_generators(C)) do c
-    Fc = ob_map(F, c)
-    colimit(compose(op(Fc), y))
+    Fc = ob_map(F, c) # e.g. I
+    clim_diag = deepcopy(compose(op(Fc), y))
+    # modify the diagram we take a colimit of to concretize some vars
+    params = Fc isa SimpleDiagram ? Dict() : Fc.params
+    G = dom(clim_diag.diagram).cat.graph
+    for (i,val) in collect(params)
+      v = add_vertex!(G; vname=Symbol("param$i"))
+      add_edge!(G, v, i; ename=Symbol("param$i"))
+      at = nameof(ob_map(Fc, i)) # attribute type name 
+      h = only(homomorphisms(ob_map(clim_diag,i), ACS(); initial=Dict(at=>[val])))
+      push!(clim_diag.diagram.ob_map, ACS())
+      push!(clim_diag.diagram.hom_map, h)
```

Comment:
These private fields are not supposed to be mutable. Rather than deep copying the diagram `clim_diag` and mutating it, we should copy the indexing graph and mutate that, then construct a new diagram using that.